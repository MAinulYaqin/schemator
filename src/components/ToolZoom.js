import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Tooltip from "./Tooltip";

const StyledTooltip = styled(Tooltip)``;

const Container = styled.div`
  width: 64px;
  height: 32px;
  box-sizing: border-box;
  margin: 4px;
  display: flex;
  position: relative;

  &:hover ${StyledTooltip} {
    opacity: 1;
  }
`;

export const Select = styled.select`
  width: 64px;
  height: 32px;
  border-radius: 4px;
  border: 0px;
  background: #444;
  color: white;
  outline: none;
  cursor: pointer;

  &:disabled {
    color: #777;
  }
`;

const ToolZoom = ({ value, tooltip, isDisabled, onChange }) => {
  const items = [25, 33, 50, 67, 75, 80, 90, 100, 110, 125, 150, 175, 200];

  return (
    <Container>
      {tooltip && <StyledTooltip text={tooltip} />}
      <Select value={value} onChange={onChange} disabled={isDisabled}>
        {items.map((item, index) => (
          <option key={index} value={item}>
            {item}%
          </option>
        ))}
      </Select>
    </Container>
  );
};

ToolZoom.propTypes = {
  value: PropTypes.func,
  tooltip: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool
};

ToolZoom.defaultProps = {
  value: 100
};

export default ToolZoom;
