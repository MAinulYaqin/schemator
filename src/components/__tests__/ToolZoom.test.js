import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import ToolZoom, { Select } from "../ToolZoom";

describe("ToolZoom", () => {
  const setup = propOverrides => {
    const props = {
      value: 25,
      tooltip: "Zoom",
      onChange: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<ToolZoom {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders as disabled", () => {
    const { wrapper } = setup({ isDisabled: true });

    expect(wrapper.find(Select).props().disabled).toEqual(true);
  });

  it("should calls 'onChange' while changed", () => {
    const { wrapper, props } = setup();
    const event = {
      target: {
        value: 50
      }
    };

    wrapper.find(Select).simulate("change", event);

    expect(props.onChange).toBeCalledWith(event);
  });
});
