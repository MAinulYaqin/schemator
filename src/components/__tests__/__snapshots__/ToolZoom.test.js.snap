// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolZoom should renders default 1`] = `
<styled.div>
  <Styled(Tooltip)
    text="Zoom"
  />
  <styled.select
    onChange={[Function]}
    value={25}
  >
    <option
      key="0"
      value={25}
    >
      25
      %
    </option>
    <option
      key="1"
      value={33}
    >
      33
      %
    </option>
    <option
      key="2"
      value={50}
    >
      50
      %
    </option>
    <option
      key="3"
      value={67}
    >
      67
      %
    </option>
    <option
      key="4"
      value={75}
    >
      75
      %
    </option>
    <option
      key="5"
      value={80}
    >
      80
      %
    </option>
    <option
      key="6"
      value={90}
    >
      90
      %
    </option>
    <option
      key="7"
      value={100}
    >
      100
      %
    </option>
    <option
      key="8"
      value={110}
    >
      110
      %
    </option>
    <option
      key="9"
      value={125}
    >
      125
      %
    </option>
    <option
      key="10"
      value={150}
    >
      150
      %
    </option>
    <option
      key="11"
      value={175}
    >
      175
      %
    </option>
    <option
      key="12"
      value={200}
    >
      200
      %
    </option>
  </styled.select>
</styled.div>
`;
